scarb/src/compiler/compilers/starknet_contract.rs

```diff
@@ -297,32 +306,40 @@ fn find_project_contracts(
         find_contracts(db, &main_crate_ids)
     };
 
-    let external_contracts = if let Some(external_contracts) = external_contracts {
-        let _ = trace_span!("find_external_contracts").enter();
-        debug!("external contracts selectors: {:?}", external_contracts);
-
-        let crate_ids = external_contracts
-            .iter()
-            .map(|selector| selector.package().into())
-            .unique()
-            .map(|package_name: SmolStr| {
-                db.upcast_mut()
-                    .intern_crate(CrateLongId::Real(package_name))
-            })
-            .collect::<Vec<_>>();
-        find_contracts(db, crate_ids.as_ref())
-            .into_iter()
-            .filter(|decl| {
-                external_contracts.iter().any(|selector| {
+    let external_contracts: Vec<ContractDeclaration> =
+        if let Some(external_contracts) = external_contracts {
+            let _ = trace_span!("find_external_contracts").enter();
+            debug!("external contracts selectors: {:?}", external_contracts);
+
+            let crate_ids = external_contracts
+                .iter()
+                .map(|selector| selector.package().into())
+                .unique()
+                .map(|package_name: SmolStr| {
+                    db.upcast_mut()
+                        .intern_crate(CrateLongId::Real(package_name))
+                })
+                .collect::<Vec<_>>();
+            let contracts = find_contracts(db, crate_ids.as_ref());
+            let filtered_contracts: Vec<ContractDeclaration> = contracts
+                .into_iter()
+                .filter(|decl| {
                     let contract_path = decl.module_id().full_path(db.upcast());
-                    contract_path == selector.full_path()
+                    external_contracts.iter().any(|selector| {
+                        if selector.is_wildcard() {
+                            contract_path.starts_with(&selector.package_path())
```

Comment:
I do not think this logic is correct. 
As of now, selector with `some::cairo::path::*` would match all contracts in `some` package.

Reply:
Please also add a test case for this scenario.

Reply:
change package_path to partial_path selector with modified logic